### DISCUSSION TITLE
Is there any way to output non expanded strings when using PowerShell to read Windows environment variables?

### DISCUSSION THREAD

**Discussion by Curious-r** [PowerShell/PowerShell#17518] [Q&A] [answered]
As the title says，I want to add PATH in Windows with PowerShell, however, when reading  existing PATH, entries like `%SystemRoot%\system32` will be expanded to `C:\WINDOWS\system32`. This will cause the existing entries to be changed when a new entry is added and written back to PATH. 
So is there anyone can help me?

**Comment by jborean93** [comment, marked as answer]
You need to call `GetValue()` on the registry key you want and read it with a special flag to get the un-expanded strings.

```powershell
(Get-Item -Path "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment").GetValue(
    "PATH",
    "",
    [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames)
```

**Reply by Curious-r** [reply]
Great!!! This is what I need, thanks a lot for your help! 😁
